web/src/main/java/ua/com/vertex/controllers/ImageController.java

```diff
@@ -7,57 +7,74 @@
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.multipart.MultipartFile;
 import ua.com.vertex.beans.User;
 import ua.com.vertex.logic.interfaces.UserLogic;
 
-import java.sql.SQLException;
+import static ua.com.vertex.controllers.exceptionHandling.GlobalExceptionHandler.ERROR_MESSAGE;
 
 @Controller
 public class ImageController {
     private static final Logger LOGGER = LogManager.getLogger(ImageController.class);
-    private static final String USER_PROFILE = "userProfile";
-    private static final String PAGE_TO_DISPLAY = "pageToDisplay";
     private static final String IMAGE = "image";
+    private static final String IMAGE_PHOTO = "imagePhoto";
+    private static final String IMAGE_PASSPORT = "imagePassport";
     private static final String IMAGE_TYPE = "imageType";
     private static final String USER = "user";
+    private static final String INDEX = "index";
     private static final String IMAGE_ERROR = "imageError";
 
     private final UserLogic userLogic;
 
-    @GetMapping(value = "/showImage")
-    public String showImage(@ModelAttribute(USER) User user,
-                            @RequestParam(PAGE_TO_DISPLAY) String pageToDisplay,
-                            @RequestParam(IMAGE_TYPE) String imageType, Model model) throws SQLException {
+    @PostMapping(value = "/showImagePhoto")
+    public String showImagePhoto(@ModelAttribute(USER) User user,
+                                 @RequestParam(IMAGE_TYPE) String imageType, Model model) {
 
-        LOGGER.debug(pageToDisplay + " page accessed");
-        encode(model, user.getUserId(), imageType);
+        LOGGER.debug(IMAGE_PHOTO + " page accessed");
+        encode(model, user.getEmail(), imageType);
         model.addAttribute(USER, user);
-        LOGGER.debug("Passing image to JSP");
+        LOGGER.debug("Passing imagePhoto to JSP");
 
-        return pageToDisplay;
+        return IMAGE_PHOTO;
     }
 
-    private void encode(Model model, int userId, String imageType) throws SQLException {
-        String encoded = Base64.encodeBase64String(userLogic.getImage(userId, imageType).orElse(new byte[]{}));
+    @PostMapping(value = "/showImagePassport")
+    @PreAuthorize("(principal.username).equals(#user.email) || hasRole('ADMIN')")
+    public String showImagePassport(@ModelAttribute(USER) User user,
+                                    @RequestParam(IMAGE_TYPE) String imageType, Model model) {
+
+        LOGGER.debug(IMAGE_PASSPORT + " page accessed");
+        encode(model, user.getEmail(), imageType);
+        model.addAttribute(USER, user);
+        LOGGER.debug("Passing imagePassport to JSP");
+
+        return IMAGE_PASSPORT;
+    }
+
+    private void encode(Model model, String email, String imageType) {
+        String encoded = Base64.encodeBase64String(userLogic.getImage(email, imageType).orElse(new byte[]{}));
         model.addAttribute(imageType, encoded);
```

Comment:
model is indirectly changed. Can you please get encoded image from the method and set it to the model inside interface method.

Reply:
Done